WordPress/Classes/ViewRelated/Comments/CommentViewController.m

```diff
@@ -85,6 +88,15 @@ - (void)loadView
         [self.tableView registerNib:tableViewCellNib forCellReuseIdentifier:[cellClass reuseIdentifier]];
     }
 
+    // If the current user cannot moderate the comment, they can only Like and Reply if the comment is Approved.
+    if (self.comment.blog.isHostedAtWPcom &&
+        !self.comment.canModerate &&
+        ![self.comment.status isEqualToString:CommentStatusApproved]) {
```

Comment:
I think you can replace:
```objc
[self.comment.status isEqualToString:CommentStatusApproved])
```
with:
```objc
[self.comment.isApproved]
```
---

Additionally, would it be better to move this to the `Comment` model? e.g.
```objc
self.userCanLikeAndReply = !self.comment.isReadOnly;
```

Reply:
> [self.comment.isApproved]

Cool, I didn't notice that.

> Additionally, would it be better to move this to the Comment model?

Good idea!

Both are done. Thanks @dvdchr !